src/controllers/userController.js

```diff
@@ -0,0 +1,130 @@
+import moment from 'moment';
+import crypto from 'crypto';
+import Validation from '../validation';
+import { sendResetMail, sendSignupMail } from '../services/sendMail';
+import { errorResponse, successResponse } from '../utils/response';
+import Hash from '../utils/hash';
+import models from '../models';
+
+const { User, Login, Reset } = models;
+
+export default class UserController {
+  /**
+   * @description Generate link to reset a user password
+   * @static
+   * @param {*} req
+   * @param {*} res
+   * @returns Promise {UserController} A reset link for new password
+   * @memberof UserController
+   */
+  static async forgotPassword(req, res) {
+    try {
+      const { errors, isValid } = Validation.validateEmail(req.body);
+
+      // Check validation
+      if (!isValid) {
+        return errorResponse(res, 400, errors);
+      }
+
+      const { email } = req.body;
+
+      // Find user by email
+      req.user = await User.findOne({ where: { email } });
+      const { user } = req;
+      // Check for user
+      if (!user) {
+        sendSignupMail(email);
+      } else {
+        const newReset = new Reset({
+          id: user.id,
+          email: req.body.email,
+          reset_token: '',
+          created_on: new Date(),
+          expire_time: moment
+            .utc()
+            .add(process.env.TOKENEXPIRY, 'seconds')
+            .toLocaleString()
+        });
+
+        // Generate Reset token
+        const resetToken = await crypto.randomBytes(32).toString('hex');
+        newReset.reset_token = await Hash.hash(resetToken);
+        // Remove all reset token for this user if it exists
+        await Reset.destroy({
+          where: { email: newReset.dataValues.email }
+        });
+        await newReset.save();
+        // Send reset link to user email
+        await sendResetMail(user.dataValues, resetToken);
+      }
+      successResponse(res, 200, 'Check your mail for further instruction');
+    } catch (error) {
+      return errorResponse(res, 500, error);
+    }
+  }
+
+  /**
+   * @description Resets a user password
+   * @static
+   * @param {*} req
+   * @param {*} res
+   * @returns Promise {UserController} A new password record
+   * @memberof UserController
+   */
+  static async resetPassword(req, res) {
+    try {
+      const { errors, isValid } = Validation.validatePassword(req.body);
+
+      // Check validation
+      if (!isValid) {
+        if (errors.password && errors.password === 'Passwords must match') {
+          return errorResponse(res, 409, errors);
```

Comment:
Why is the res status 409?

Reply:
> Why is the res status 409?

409 stands for conflict, i realize now that the correct status should be 401, "request has not been applied because it lacks valid authentication credentials for the target resource". I have applied the correct status code.